examples/msr-events.rs

```diff
@@ -57,7 +56,7 @@ fn toggle_msr_intercepts(drv: &mut Box<dyn Introspectable>, vec_msr: &Vec<u32>,
         println!("{} intercept on 0x{:x}", status_str, msr);
         for vcpu in 0..drv.get_vcpu_count().unwrap() {
             drv.toggle_intercept(vcpu, intercept, enabled)
-                .expect(&format!("Failed to enable 0x{:x}", msr));
+                .unwrap_or_else(|_| panic!("Failed to enable 0x{:x}", msr));
```

Comment:
Is this some weird tactic to silence static analysis warnings? I don't really understand what's going on here.

Reply:
This is the clippy warning:
~~~rust
warning: use of `expect` followed by a function call
  --> examples/msr-events.rs:59:18
   |
59 |                 .expect(&format!("Failed to enable 0x{:x}", msr));
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| panic!("Failed to enable 0x{:x}", msr))`
   |
   = note: `#[warn(clippy::expect_fun_call)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 1.77s
~~~

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call